src/actions/notification-builder.js

```diff
@@ -160,27 +190,61 @@ class NotificationBuilder {
   /**
    * Sends the notification.
    *
+   * If no reference ID is set, a random reference ID is generated.
    * In case the openHAB Cloud Connector is not installed, a warning is logged and the notification is not sent.
+   *
+   * @return {string|null} the reference ID of the notification or `null` log notifications and when hiding notifications
+   * @throws {Error} if {@link hide} was called and no reference ID or tag is set
    */
   send () {
     while (this.#actionButtons.length < 3) {
       this.#actionButtons.push(null);
     }
+
     switch (this.#type) {
       case NotificationType.BROADCAST:
-        // parameters: message, icon, severity, title, onClickAction, mediaAttachmentUrl, actionButton1, actionButton2, actionButton3
-        _getNotificationAction()?.sendBroadcastNotification(this.#message, this.#icon, this.#severity, this.#title, this.#onClickAction, this.#mediaAttachmentUrl, ...this.#actionButtons);
-        break;
+        if (this.#referenceId === null) this.#referenceId = randomUUID();
```

Comment:
Any reason for not just letting the referenceId be null instead of setting it to  random value?

Reply:
also  now when i run it, i'm getting this error. I think this is b/c this is setting to a `java.util.UUID` object  and not an actual string. 

```
Failed to execute script: TypeError: invokeMember (sendBroadcastNotification) on org.openhab.io.openhabcloud.NotificationAction failed due to: no applicable overload found (overloads: [Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String)], Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String,java.lang.String,java.lang.String)], Method[public static void org.openhab.io.openhabcloud.NotificationAction.sendBroadcastNotification(java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String,java.lang.String)]], arguments: [Test Message (String), f7:bell_fill (String), important (String), Important Notification (String), JavaObject[b6010bfd-b2f6-4e96-bc76-cb241287de7c (java.util.UUID)] (HostObject), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish), DynamicObject<null>@3c2604c8 (Nullish)])
```

Reply:
Ah i see what you are doing, very cleaver! You are returning the ID for use later by a rule.  Nice 👍  i did not put together we could do that now.